src/netlink_pm_upstream.c

```diff
@@ -367,17 +367,27 @@ static int upstream_add_addr(struct mptcpd_pm *pm,
           Payload (nested):
               Local address family
               Local address
+              Local port (optional)
               Local address ID (optional)
               Flags (optional)
               Network inteface index (optional)
          */
 
         // Types chosen to match MPTCP genl API.
         uint16_t const family = mptcpd_get_addr_family(addr);
+        uint16_t const port   = get_port_number(addr);
+
+        /*
+          The MPTCP_PM_ADDR_FLAG_SIGNAL flag is required when a port
+          is specified.  Make sure it is set.
+        */
+        if (port != 0)
```

Comment:
This assumes that a port value of zero implies that the port is not specified.  Is that reasonable?  Does is make sense to pass a port value of zero to the kernel?

Reply:
Thanks for confirming the kernel side behavior when given a zero port.

Regarding `MPTCP_PM_ADDR_FLAG_SIGNAL`, is there a case where one wouldn't use `MPTCP_PM_ADDR_FLAG_SIGNAL` with a non-zero port.  Assuming there isn't, other than making the user aware of the signaling, it seems odd to force the user to set the flag each time a non-zero port is used.  Is setting the flag transparently too "magic"?

Reply:
We should probably differentiate between the "user" and "plugin authors". It can be reasonable to have detailed rules for plugin authors, and we can make that easier with warnings and such. In this case, I think we're talking about what to ask of plugin authors.

I suggest looking at `man ip mptcp` for the flag descriptions:

```
       signal the endpoint will be announced/signalled to each peer via an
              ADD_ADDR MPTCP sub-option

       subflow
              if additional subflow creation is allowed by MPTCP limits, the
              endpoint will be used as the source address to create an addi‐
              tional subflow after that the MPTCP connection is established.

       backup the endpoint will be announced as a backup address, if this is a
              signal endpoint, or the subflow will be created as a backup one
              if this is a subflow endpoint
```
Looking at that description, and the example commands in various documentation, I don't see where an empty flags value is used anywhere. Might make more sense to handle the differentiation between "announce address and accept incoming joins (signal)" and "send outgoing join (subflow)" some other way in the API, handle the signal/subflow flags transparently at this layer, and use a flag for `backup`? Or maybe `backup` is its own parameter and we get rid of the open-ended flags. I'm not super sure of the answer here, so let's discuss!

Reply:
I think you're right.  Rather than expose the PM netlink API idiosyncrasies to the plugins it would be better to provide plugins with a cleaner and simpler API or abstraction.